api/v1/controllers/collectors.js

```diff
@@ -320,18 +320,75 @@ function heartbeat(req, res, next) {
 } // heartbeat
 
 /**
- * Change collector status to Running. Invalid if the collector's status is
- * not Stopped.
+ * Creates a collector if name not found, with the user as the sole writer.
+ * Change collector status to Running and returns a new collector token.
+ * Reject if the user is not among the writers.
+ * Invalid if the collector's status is not Stopped.
  *
  * @param {IncomingMessage} req - The request object
  * @param {ServerResponse} res - The response object
  * @param {Function} next - The next middleware function in the stack
  */
 function startCollector(req, res, next) {
-  req.swagger.params.queryBody = {
-    value: { status: 'Running' },
-  };
-  doPatch(req, res, next, helper);
+  const toPost = req.swagger.params.queryBody.value;
+  toPost.status = 'Running';
+  let collector;
+
+  // returns null if no collector found
+  return helper.model.findOne({ where: { name: toPost.name } })
+  .then((returnedCollector) => {
+    if (!returnedCollector) {
+
+      // create collector, find user
+      const username = jwtUtil.getTokenDetailsFromRequest(req).username;
+      return Promise.all([
+        helper.model.create(toPost),
+        userProps.model.findOne({
+          where: {
+            name: username,
+          },
+        }),
+      ])
+      .then((results) => {
+        collector = results[0];
+        return collector.addWriter(results[1]);
```

Comment:
adding writers should be done in the DB hooks and it has already been done there.

Reply:
good point. will remove.